packages/react-noop-renderer/src/createReactNoop.js

```diff
@@ -1142,20 +1142,33 @@ function createReactNoop(reconciler: Function, useMutation: boolean) {
 
       function logUpdateQueue(updateQueue: UpdateQueue<mixed>, depth) {
         log('  '.repeat(depth + 1) + 'QUEUED UPDATES');
-        const firstUpdate = updateQueue.firstUpdate;
-        if (!firstUpdate) {
+        const last = updateQueue.baseQueue;
+        if (last === null) {
           return;
         }
+        const first = last.next;
+        let update = first;
+        if (update !== null) {
+          do {
+            log(
+              '  '.repeat(depth + 1) + '~',
+              '[' + update.expirationTime + ']',
+            );
+          } while (update !== null && update !== first);
```

Comment:
Can `update` be changed while inside the loop? Do we need to additionally check for not null being inside `if (update !== null)`?

Reply:
Good catch. It's missing a `update = update.next` line at the end of the loop. Honestly I don't think anyone actually uses this method anymore, so I'll probably just remove it.

Reply:
Yeah, I believe removal would be safe here and in the block below.